File: app/src/main/java/com/travlog/android/apps/libs/qualifiers/RequiresFragmentViewModel.java
package com.travlog.android.apps.libs.qualifiers;

import com.travlog.android.apps.FragmentViewModel;

import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Inherited
@Retention(RetentionPolicy.RUNTIME)
public @interface RequiresFragmentViewModel {
    Class<? extends FragmentViewModel> value();
}

File: app/src/main/java/com/travlog/android/apps/libs/qualifiers/RequiresFragmentViewModel.kt
package com.travlog.android.apps.libs.qualifiers

import com.travlog.android.apps.FragmentViewModel

import java.lang.annotation.Inherited
import kotlin.reflect.KClass

@Inherited
@Retention(AnnotationRetention.RUNTIME)
annotation class RequiresFragmentViewModel(val value: KClass<out FragmentViewModel<*>>)
